internal/rollup/terser-wrapped.js

```diff
@@ -20,47 +20,84 @@
 const fs = require('fs');
 const path = require('path');
-const tmp = require('tmp');
-const child_process = require('child_process');
+const os = require('os');
+const pLimit = require('p-limit');
+
+let worker_threads;
+let WORKER_THREADS_AVILABLE = true;
+
+try {
+  worker_threads = require('worker_threads');
+} catch (_e) {
+  WORKER_THREADS_AVILABLE = false;
+  console.warn(
+      `WARNING: worker_threads is not avilable on the version of Node.js you are running under Bazel: ${
+          process.version}
+  Optimizing one file at a time
+  Upgrade your version of Node.js to be at least 10.5.0 in your WORKSPACE file for faster builds with worker_threads`)
+}
 
 const DEBUG = false;
+const TERSER_MODULE_LOCATION = 'build_bazel_rules_nodejs_rollup_deps/node_modules/terser';
+
+function main() {
+  // capture the inputs and output options
+  const argv = require('minimist')(process.argv.slice(2));
+  const inputs = argv._;
+  const output = argv.output || argv.o;
+  const debug = argv.debug;
+
+  if (DEBUG)
+    console.error(`
+  terser: running with
+    cwd: ${process.cwd()}
+    argv: ${process.argv.slice(2).join(' ')}
+    inputs: ${JSON.stringify(inputs)}
+    output: ${output}
+    debug: ${debug}
+    worker_threads: ${WORKER_THREADS_AVILABLE}
+  `);
+
+  if (inputs.length != 1) {
+    throw new Error(`Only one input file supported: ${inputs}`);
+  }
 
-// capture the inputs and output options
-const argv = require('minimist')(process.argv.slice(2));
-const inputs = argv._;
-const output = argv.output || argv.o;
-const debug = argv.debug;
-const configFile = argv['config-file'];
-// delete the properties extracted above as the remaining
-// arguments are forwarded to terser in execFileSync below
-delete argv._;
-delete argv.output;
-delete argv.o;
-delete argv.debug;
-delete argv['config-file'];
-
-if (DEBUG)
-  console.error(`
-terser: running with
-  cwd: ${process.cwd()}
-  argv: ${process.argv.slice(2).join(' ')}
-  inputs: ${JSON.stringify(inputs)}
-  output: ${output}
-  debug: ${debug}
-`);
-
-if (inputs.length != 1) {
-  throw new Error(`Only one input file supported: ${inputs}`);
-}
+  const input = inputs[0];
 
-const input = inputs[0];
+  const isDirectory = fs.lstatSync(path.join(process.cwd(), input)).isDirectory();
 
-function runterser(inputFile, outputFile, sourceMapFile) {
-  if (DEBUG) console.error(`Minifying ${inputFile} -> ${outputFile} (sourceMap ${sourceMapFile})`);
+  if (!isDirectory) {
+    runTerser(input, output, output + '.map', require(TERSER_MODULE_LOCATION));
+  } else {
+    if (!fs.existsSync(output)) {
+      fs.mkdirSync(output);
+    }
+    const dir = fs.readdirSync(input);
+    const files = dir.filter(f => f.endsWith('.js')).map(f => {
+      const inputFile = path.join(input, path.basename(f));
+      const outputFile = path.join(output, path.basename(f));
+      return {inputFile, outputFile, sourceMapFile: outputFile + '.map'};
+    })
+
+    // TODO: allow config to force serial workers here
```

Comment:
@Globegitter Suggested this one.
It would be a good addition but probably not apart of this first cut.